ntex/src/framed/state.rs

```diff
@@ -51,35 +51,13 @@ pub(crate) struct IoStateInner {
     read_task: LocalWaker,
     write_task: LocalWaker,
     dispatch_task: LocalWaker,
-    read_buf: Cell<Option<BytesMut>>,
-    write_buf: Cell<Option<BytesMut>>,
+    pool: P,
```

Comment:
I don't like idea of passing pool everywhere. should we use static methods instead , so we could do `P::take()` instead of `pool.take()`

Reply:
No. There are multiple pools in play, and they cannot be assumed to be split via `task_local!` or `thread_local!` either.

Reply:
why do need of multiple pools of same type?

Reply:
Flexibility. Simplest example is that we want the ability to have different QoS for reading vs writing, which means `Decoder` and `Encoder` would be using different pools.

Same type does not mean the pools are identical. As I'd said in the original email, [our pool](https://github.com/Azure/iotedge-broker/blob/46147f14c2dcc775bc321ca169ac67b1d6f20d42/data-plane/buffer-pool/src/custom.rs#L22-L36) has configuration to change the size of static tiers and dynamic tiers. So different pools of the same type can have different size of buffers and dynamic capacity.

Reply:
if you need different strategies for read and write then you need to pass 2 pools here

Reply:
Yes. As I said in PR description, changes to `ntex::*` are not done yet.